### PR TITLE
Fixed rubocop issues, new .rubocop_todo.yml

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
- fixed new rubocop issues
- generated a new `.rubocop_todo.yml`

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -54,9 +67,9 @@ Style/GuardClause:
   Exclude:
     - 'app/models/article.rb'
 
-# Offense count: 2101
+# Offense count: 2415
```
Comment (rhymes) [inline .rubocop_todo.yml]: ahhaha this keeps increasing :D

File: app/controllers/notification_subscriptions_controller.rb [hunk 2]
```diff
@@ -18,9 +18,7 @@ def upsert
       @notification_subscription.notifiable.update(receive_notifications: false) if current_user_is_author?
     else
       @notification_subscription.config = params[:config] || "all_comments"
-      if params[:config] == "all_comments" && current_user_is_author?
-        @notification_subscription.notifiable.update(receive_notifications: true)
-      end
+      @notification_subscription.notifiable.update(receive_notifications: true) if params[:config] == "all_comments" && current_user_is_author?
```
Comment (rhymes) [inline app/controllers/notification_subscriptions_controller.rb]: I'm not a mega fan of the giant ifs at the end, i think the code was more readable before. I'm thinking we should not enforce this, what do you think?

File: app/liquid_tags/poll_tag.rb [hunk 3]
```diff
@@ -22,98 +22,96 @@ def find_poll(id_code)
...
+                    document.getElementById('showmethemoney-'+json.poll_id).addEventListener('click', function(e) {
+                      pollId = this.dataset.pollId
+                      window.fetch('/poll_skips', {
                         method: 'POST',
                         headers: {
                           'X-CSRF-Token': csrfToken,
                           'Content-Type': 'application/json',
                         },
-                        body: JSON.stringify({poll_vote: { poll_option_id: optionId } }),
+                        body: JSON.stringify({poll_skip: {poll_id: pollId }}),
```
Comment (rhymes) [inline app/liquid_tags/poll_tag.rb]: should this be part of this PR? It seems it's changing behavior here

File: app/controllers/notification_subscriptions_controller.rb (hunk 2 above)
Comment (lightalloy) [reply app/controllers/notification_subscriptions_controller.rb]: Agree, I haven't checked the autocorrected stuff properly.

File: app/liquid_tags/poll_tag.rb (hunk 3 above)
Comment (lightalloy) [reply app/liquid_tags/poll_tag.rb]: It's `poll_skip` and `poll_id` in master:
https://github.com/thepracticaldev/dev.to/blob/e46bd26223847e35b9de2cd57380bad12a97286f/app/liquid_tags/poll_tag.rb#L96
I wonder why it shows the changes.

Comment (rhymes) [reply app/liquid_tags/poll_tag.rb]: It's weird, have you tried to rebase?

Comment (lightalloy) [reply app/liquid_tags/poll_tag.rb]: Yep, "up-to-date".

Comment (rhymes) [reply app/liquid_tags/poll_tag.rb]: It happened to me once or twice, what I did was to create a new branch, cherry pick the commits from the old one in order and either force push or create a new PR.

I still don't know why :(

Comment (rhymes) [reply app/liquid_tags/poll_tag.rb]: BTW the second commit contains the change so cherry picking it won't work https://github.com/thepracticaldev/dev.to/pull/3255/commits/e2f683752ba3f6ea2fe21e36de142e69ed7f8783

Comment (lightalloy) [reply app/liquid_tags/poll_tag.rb]: Actually, if you look at line 81 red and green - they are identical, and the line 96 red and green - they are identical as well (apart from the indentation). 
But the `git diff` looks confusing.

Comment (benhalpern) [reply app/liquid_tags/poll_tag.rb]: Ah, that's confusing.

**Pull request merged**